include/pmacc/math/vector/Vector.hpp

```diff
@@ -22,6 +22,11 @@
 
 #pragma once
 
+#if BOOST_COMP_CLANG
+    #pragma clang diagnostic push
```

Comment:
Please also add the `pop` in this file else this flag will also effect other files.

Reply:
Ok, if we keep this commit.

I encountered the problem, that even gcc is issuing this warning:
```
 warning: ISO C++11 requires at least one argument for the "..." in a variadic macro
```
This is when using the language standard `-std=c++14`. It only goes away when using `-std=gnu++14`, which I guess is what CMake is doing by default(?).

Would there be a proper fix for this to be compliant with the ISO standard?

Reply:
Could you please post the full warning. I do not see any variadic template in this file.

Reply:
Apparently this can be made compliant with the standard by putting a comma after the last named argument to explicitly have zero `__VA_ARGS__` , [see GCC docs](https://gcc.gnu.org/onlinedocs/cpp/Variadic-Macros.html). According to this site, the currenty style will become compliant with C++2a (I did not check if it was accepted in C++20).

Would such a change of style be acceptable to get rid of this warning for good?

Reply:
The problem is not the macro but its invocation. The full warning is:
```
./include/pmacc/math/vector/Vector.hpp:89:52: warning: ISO C++11 requires at least one argument for the "..." in a variadic macro
    89 |                           T_Dest::dim == T_Src::dim);
       |                                                    ^
```
https://github.com/ComputationalRadiationPhysics/picongpu/pull/3255/commits/76cc7a437379e3a9e63fa85e8bc5ab341f8650a0 addresses the root cause of this warning.